src/par_iter/from_par_iter.rs

```diff
@@ -61,8 +96,10 @@ impl<PAR_ITER, K, V> FromParallelIterator<PAR_ITER> for BTreeMap<K, V>
           V: Send,
 {
     fn from_par_iter(par_iter: PAR_ITER) -> Self {
-        let vec: LinkedList<(K, V)> = par_iter.collect();
-        vec.into_iter().collect()
+        let list = collect_list_vec(par_iter);
```

Comment:
Nit: it feels like we can derive a helper here; something like

```rust
fn combine<...>(par_iter: PAR_ITER, make_start: FnOnce(&LinkedList<Vec<T>>)) {
    let list = collect_list_vec(par_iter);
    let start = make_start(&list);
    list.into_iter().fold(start, |mut coll, vec| { coll.extend(vec); coll })
}
```

given suitable generics and bounds :)